src/ert/cli/model_factory.py

```diff
@@ -1,111 +1,115 @@
+from __future__ import annotations
+
 import logging
-from typing import List
+from typing import TYPE_CHECKING, List
+from uuid import UUID
 
 from ert._c_wrappers.config.active_range import ActiveRange
+from ert._c_wrappers.enkf.enkf_main import EnKFMain
 from ert.shared.models.ensemble_experiment import EnsembleExperiment
 from ert.shared.models.ensemble_smoother import EnsembleSmoother
 from ert.shared.models.iterated_ensemble_smoother import IteratedEnsembleSmoother
 from ert.shared.models.multiple_data_assimilation import MultipleDataAssimilation
 from ert.shared.models.single_test_run import SingleTestRun
 
+if TYPE_CHECKING:
+    from ert.storage import StorageAccessor
+
 
-def create_model(ert, ensemble_size, current_case_name, args, id_):
+def create_model(
+    ert: EnKFMain,
+    storage: StorageAccessor,
+    args,
+    experiment_id: UUID,
+):
     logger = logging.getLogger(__name__)
     logger.info(
         "Initiating experiment",
         extra={
             "mode": args.mode,
-            "ensemble_size": ensemble_size,
+            "ensemble_size": ert.getEnsembleSize(),
```

Comment:
Both are `ert.getEnsembleSize()`. This is information for the experiment that we are running now, rather than some prior experiment that we have run in the past. As such, it makes sense to use `ert.getEnsembleSize()`, where the `EnKFMain` object is a (bad) way of storing data about the currently running experiment.